File: 2186.使两字符串互为字母异位词的最少步骤数.go
/*
 * @lc app=leetcode.cn id=2186 lang=golang
 *
 * [2186] 使两字符串互为字母异位词的最少步骤数
 */

// @lc code=start
func minSteps(s string, t string) int {
	res := 0
	str2cnt := make(map[rune]int)
	for _, v := range s {
		str2cnt[v]++
	}
	for _, v := range t {
		str2cnt[v]--
	}
	for _, v := range str2cnt {
		res += abs(v)
	return res
}
func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

// @lc code=end

File: 2186.使两字符串互为字母异位词的最少步骤数.py
#
# @lc app=leetcode.cn id=2186 lang=python3
#
# [2186] 使两字符串互为字母异位词的最少步骤数
#

# @lc code=start
class Solution:
    def minSteps(self, s: str, t: str) -> int:
        s_c = Counter(s)
        t_c = Counter(t)
        res = 0
        for key in s_c:
            if key in t_c:
                res += max(0, abs(s_c[key] - t_c[key])))
            else:
                res += s_c[key]
        for key in t_c:
            if key not in s_c:
                res += t_c[key]
        return res
# @lc code=end

